### PR TITLE
Docker refactoring

### PR DESCRIPTION
**Pull request by simison**
WIP

I've been piling up some changes to my docker setup and I thought someone might want to see these already now when they're still kinda WIP.

Mostly:
- Docker-compose has [very flexible env-support](https://docs.docker.com/compose/environment-variables/) so I'm exploring using that. I'd like to just keep my configs (mostly domain and passwords) in `.env` file which survives between WordPress reinstalls. There is `default.env` which is used by default but any values in `.env` overwrite those. You could override only e.g. domain.
- Has a script to wipe out WP (`yarn docker:uninstall`)
- Has a script to fill WP with test data (`yarn docker:fill` — be warned, it's slow)
- wp-cli is configured globally, so no need to pass `--path` all the time — sadly still needs `--allow-root`
- I moved config files to `./docker/configs`
- I moved some yarn scripts to bash files in `./docker/bin` — they're run from inside the container so Windows users 👌 I moved these to here mainly just because having these as one-liner yarn scripts started to be a bit hard to read... :-D 

Thoughts?

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -47,23 +47,23 @@ RUN curl https://phar.phpunit.de/phpunit-5.7.5.phar -L -o phpunit.phar \
 	&& mv phpunit.phar /usr/local/bin/phpunit
 
 # Copy a default config file for an apache host
-COPY ./apache_default /etc/apache2/sites-available/000-default.conf
+COPY ./configs/apache_default /etc/apache2/sites-available/000-default.conf
```
Comment (oskosk) [inline docker/Dockerfile]: `configs` (plural) sounds weird to me

Comment (simison) [reply docker/Dockerfile]: Fair enough. :-D I guess I was just looking at `docker/logs` and not at `docker/data` while typing it. ;-)

Comment (oskosk) [reply docker/Dockerfile]: Actually it does not sound weird to me, but it sounds weird in English or not used to see a dir named like that. How about `config-templates` ?

**Comment by simison** [comment]
Closing as I've merged most of the things from this exploration already.

Only major thing missing is:
> Has a script to fill WP with test data (`yarn docker:fill` — be warned, it's slow)

...but I don't think the approach in this PR is good. It's slow and relies on an external import file. There's also this cli command already: https://developer.wordpress.org/cli/commands/post/generate/

**Pull request closed without merging**